de/tests/test_factorization.py

```diff
@@ -41,3 +41,35 @@ def test_fit(sizze):
     data = np.random.lognormal(size=sizze)
     outt = runOptim(data, niter=20, disp=False)
     assert np.all(np.isfinite(outt))
+
+def test_cellLines():
+    """ To test and confirm most genes are overlapping between cell lines. """
+    cellLine1 = 'A375'
+    cellLine2 = 'HT29'
+    _, _, annotation1 = cellLineFactorization(cellLine1)
+    _, _, annotation2 = cellLineFactorization(cellLine2)
+
+    # assuming the function returns the list of shared genes between the two cell lines
+    shared_annotation, _ = cellLineComparison(cellLine1, cellLine2)
+    # make sure at least 50% of the genes in smaller cell line is shared between the two cell lines
+    assert np.abs(len(shared_annotation)) >= 0.5 * np.min([len(annotation1), len(annotation2)])
+
+def test_matrixSub():
+    """To test if the matrices subtract properly and if the norm has a reasonble value"""
+    cellLine1 = 'A375'
+    cellLine2 = 'HT29'
+    
+    _, difference_norm, _, _ = MatrixSubtraction(cellLine1, cellLine2)
+
+    w1, _, _ = cellLineFactorization(cellLine1)
+    w2, _, _ = cellLineFactorization(cellLine2)
+    np.random.shuffle(w1)
+    np.random.shuffle(w2)
+    test_norm1 = np.linalg.norm(w1)
+    test_norm2 = np.linalg.norm(w2)
+
+    assert difference_norm < test_norm1
+    assert difference_norm < test_norm2
+
+
+    
```

Comment:
please remove the empty lines. You should only have one line after the end of your last function.